pkg/visitors/insert_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package visitors
+
+import (
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/linuxboot/fiano/pkg/guid"
+	"github.com/linuxboot/fiano/pkg/uefi"
+)
+
+func TestInsert(t *testing.T) {
+	var tests = []struct {
+		name string
+		InsertType
+	}{
+		{InsertFront.String(), InsertFront},
+		{InsertEnd.String(), InsertEnd},
+		{InsertIn.String(), InsertIn},
+		{InsertAfter.String(), InsertAfter},
+		{InsertBefore.String(), InsertBefore},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			f := parseImage(t)
+
+			file, err := ioutil.ReadFile("../../integration/roms/testfile.ffs")
```

Comment:
Would it be a good idea to dedup the repetition in the tests?

Reply:
maybe, but I did this so that I would always get a new clean copy of the file. I could instead read once and do 5 copies. What do you think?

Reply:
Sorry, no what I mean is the there's ~15 lines in each of these three functions which are the same:

```
			f := parseImage(t)
 			file, err := ioutil.ReadFile("../../integration/roms/testfile.ffs")
			if err != nil {
				t.Fatal(err)
			}
 			ffs, err := uefi.NewFile(file)
			if err != nil {
				t.Fatal(err)
			}
			// Apply the visitor.
			insert := &Insert{
				Predicate:  FindFileGUIDPredicate(*testGUID),
				NewFile:    ffs,
				InsertType: test.InsertType,
			}
```

It can be refactored into a function.